src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -116,5 +117,58 @@ public class SecurityTokenDescriptor
         /// </summary>
         [DefaultValue(true)]
         public bool IncludeKeyIdInHeader { get; set; } = true;
+
+        private static int s_maxActorChainLength = 5;
+        /// <summary>
+        /// Gets or sets the maximum depth allowed when processing nested actor tokens.
+        /// This prevents excessive recursion when handling deeply nested actor tokens.
+        /// The value must be at least 0. Value 0 would mean that the actor token is not allowed to be nested.
+        /// Default value is 5. Max value is also 5
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if the value is less than 0.</exception>
+        public static int MaxActorChainLength
+        {
+            get => s_maxActorChainLength;
+            set
+            {
+                if (value < 0)
+                    throw LogHelper.LogExceptionMessage(
+                    new ArgumentOutOfRangeException(
+                    LogHelper.FormatInvariant(
+                    LogMessages.IDX11027,
+                    LogHelper.MarkAsNonPII("MaxActorChainLength"))
+                    + ". Permissible values are integers in range 0 to 5"));
+
+                s_maxActorChainLength = value;
+            }
+        }
+
+        private static string s_actoryClaimTypeName = "act";
+        /// <summary>
+        /// Gets or sets the claim type name for the actor claim.
+        /// Permissible values are 'act' or 'actort'.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if the value is null.
+        /// </exception>
+        /// <exception cref="SecurityTokenException">
+        /// Thrown if the value is not 'act' or 'actort'.
+        /// </exception>
+        public static string ActorClaimTypeName
```

Comment:
ClaimType and ClaimName in my understanding have the same meaning.
If no value is provided, considering defaulting to "act" without limiting the name that can be used. This would make the DevEx simpler while keeping it flexible.